Reflect latest changes in synthetic source documentation (#109501)
elastic/elasticsearch: docs/reference/mapping/fields/source-field.asciidoc

_fetch_ requests, like <<docs-get,get>> or <<search-search,search>>.

> If disk usage is important to you then have a look at

Reviewer: ```suggestion
If disk usage is important to you then consider the following options:
```

> <<synthetic-source,synthetic `_source`>> which shrinks disk usage at the cost of slower fetches or (not recommended)

Reviewer 2: Maybe rename `at the cost of slower fetches or (not recommended)` into `at the cost of slower access to _source, in get and search APIs`

> <<synthetic-source,synthetic `_source`>> which shrinks disk usage at the cost of slower fetches or (not recommended)
> <<disable-source-field,disabling the `_source` field>> which also shrinks disk
> usage but disables many features.

Reviewer: consider defining these briefly up here. if you don't want to define them here, I'd remove the downsides and keep the links only.

```suggestion
- Using <<synthetic-source,synthetic `_source`>>, which reconstructs source content at the time of retrieval instead of storing it on disk. This shrinks disk usage, at the cost of slower access to `_source` in <<docs-get,Get>> and <<search-search,Search>> queries.
- <<disable-source-field,Disabling the `_source` field completely>>. This shrinks disk
usage but disables features that rely on `_source`.
```
